### ISSUE TITLE
Voters don't receive emails if they don't have a name in the upload file

### ISSUE THREAD

**Issue by sigmavirus24** [benadida/helios-server#197]
Hi there,

I just wrapped up an election where we added approximately 118 voters after the fact. Each time I attempted to email voters who had not received a ballot, or picked the voter out specifically for a email, none of these 118 received a ballot. This is extremely odd, but I'm willing to provide more information as desired privately. To be brief, the voters file had the appropriate structure, but looked like:

```
4d0b09183d2f4158a20c4fa67376a07c,auser@example.com,
```

**Comment by shirlei** [comment]
Hi @sigmavirus24 have you seen https://groups.google.com/forum/#!msg/helios-voting/hR-i_FG0gI0/5gQIXghEBgAJ and https://groups.google.com/forum/#!searchin/helios-voting/celeryd%7Csort:date/helios-voting/3cgoNvC5wmg/v3Tx4owmFQAJ ?

**Comment by sigmavirus24** [comment]
@shirlei does it matter that we're using the hosted version of Helios? vote.heliosvoting.org? We're not self-hosting Helios

**Comment by shirlei** [comment]
@sigmavirus24 oh I see ... well , giving a quick look at the code, despite the fact it was possible to load the file without the third argument (name ) , it won't work to send emails. See https://github.com/shirlei/helios-server/blob/master/helios_auth/auth_systems/password.py , line 116. Send message will fail because name will be empty and user_info.get will throw 'NoneType' object has no attribute 'get'
I strongly suggest you to load a complete file... maybe using some info you have on the voter to fill up the third argument.
@benadida any thoughts on that?

**Comment by sigmavirus24** [comment]
@shirlei that confirms my theory as to why these folks didn't receive a ballot. Thank you

**Comment by sigmavirus24** [comment]
It's my opinion that users shouldn't need to have a full name in order to be able to vote, but that's a side topic

**Comment by sigmavirus24** [comment]
Alternatively, some error saying "The voter doesn't have a valid name" would be excellent. Even if you process the rest, leting me know which were defective would be a huge improvement. I'd be happy to find time to fix that if that's desirable to the project

**Comment by benadida** [comment]
I'm sorry I didn't notice this bug till now! Just so you know, you can just repeat the user's email address as their name. I hope this helps!

**Comment by sigmavirus24** [comment]
We're actually just doing https://github.com/python/psf-election/blob/bcc3385bd1ecec31f23de27f1648049fc4626f77/convert-exported-csv-for-helios.py#L55 now so that we don't have an issue.

I still think those voters shouldn't be "accepted" because it gives the election administrator a false sense of "working"

**Comment by benadida** [comment]
agreed, I see now that this is indeed a bug. I'm going to reopen and rename.

**Comment by sigmavirus24** [comment]
So potential solutions:

- Separate out users without a name and allow a re-upload of a CSV with names to update that
- Allow users to not have a name
- Outright reject the upload if any voters don't have a name
- Only reject voters without a name and provide a helpful explanation to administrators

I'd love to hear other potential solutions

**Comment by rdenadai** [comment]
Really like the ideia of re-upload a CSV file, to update existing users info, and add new ones... a simple delete button could make a case when the delete some users are a need.

**Comment by rachellawson** [comment]
Outright reject the upload seems the safest option, along with a message explaining why. The uploader can then decide how to manipulate their upload so that it contains valid data

**Comment by benadida** [comment]
closed in #446
